source/ui/dataman/AMScanThumbnailGridViewItemDelegate.h

```diff
@@ -0,0 +1,57 @@
+#ifndef AMSCANTHUMBNAILGRIDVIEWITEMDELEGATE_H
+#define AMSCANTHUMBNAILGRIDVIEWITEMDELEGATE_H
+
+#include <QItemDelegate>
+#include <QPainter>
+
+/**
+ * Delegate class for painting the items within an AMScanThumbnailGridView
+ */
+class AMScanThumbnailGridViewItemDelegate : public QItemDelegate
+{
+	Q_OBJECT
+public:
+	/**
+	 * Creates an instance of an AMScanThumbnailGridViewItemDelegate
+	 * @param parent
+	 */
+	explicit AMScanThumbnailGridViewItemDelegate(QObject *parent = 0);
+
+	/// Paints the data associated with the provided index according to options stored in the given
```

Comment:
Mentioned this to David this morning and he added that due to this being possibly involved in an inheritance chain in the future it will need to explicitly specify a virtual destructor. In future I'll ensure all such classes have explicit virtual destructors even in cases where they're not owning resources.